app/main.py

```diff
@@ -1,3 +1,17 @@
+import functools
+
+
 def cache(func):
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        key = f"{func.__name__}{args}"
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+        else:
+            cache_dict.update({key: func(*args)})
+            print("Calculating new result")
+            return func(*args)
```

Comment:
You calculate the function result a second time here (the first time is in the 14th line). 
Use the stored result in the `cache_dict`